src/python/pants/scm/git.py

```diff
@@ -117,6 +117,7 @@ def __init__(self, binary='git', gitdir=None, worktree=None, remote=None, branch
     self._remote = remote
     self._branch = branch
 
+  @property
```

Comment:
The superclass uses `@property`. We didn't detect the difference because it seems we always use Git in the wild and our original code referred to this value as `current_rev_identifier()`, so Git worked fine. But this fixes it for other SCMs.